service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -16,67 +17,72 @@
+            MetaDbPatient patient = patientMetadataRepository.findPatientByInvestigatorId(
+                    updatedMetaDbSample.getPatient().getInvestigatorPatientId());
             if (patient != null) {
-                updatedSampleManifestEntity.setPatientUuid(patient.getUuid());
+                updatedMetaDbSample.setPatientUuid(patient.getUuid());
             }
-            sampleManifestRepository.save(updatedSampleManifestEntity);
+            metaDbSampleRepository.save(updatedMetaDbSample);
         } else {
-            updatedSampleManifestEntity.setUuid(foundSample.getUuid());
-            sampleManifestRepository.updateSampleManifestJson(
-                    updatedSampleManifestEntity.getSampleManifestJsonEntity(), foundSample.getUuid());
+            foundSample.addSampleManifest(updatedMetaDbSample.getSampleManifestList().get(0));
+            metaDbSampleRepository.save(foundSample);
         }
 
-        return updatedSampleManifestEntity;
+        return updatedMetaDbSample;
     }
 
     @Override
-    public SampleManifestEntity setUpSampleManifest(SampleManifestEntity
-            sampleManifestEntity) throws Exception {
-        PatientMetadata patient = new PatientMetadata();
+    public MetaDbSample setUpMetaDbSample(MetaDbSample
+            metaDbSample) throws Exception {
+        metaDbSample = setUpSampleManifestEntity(metaDbSample);
+        SampleManifestEntity sampleManifestEntity = metaDbSample.getSampleManifestList().get(0);
+        MetaDbPatient patient = new MetaDbPatient();
         patient.setInvestigatorPatientId(sampleManifestEntity.getCmoPatientId());
-        sampleManifestEntity.setPatient(patient);
+        metaDbSample.setPatient(patient);
 
         SampleAlias igoId = new SampleAlias();
         igoId.setIdSource("igoId");
         igoId.setSampleId(sampleManifestEntity.getIgoId());
-        sampleManifestEntity.addSample(igoId);
+        metaDbSample.addSample(igoId);
 
         SampleAlias investigatorId = new SampleAlias();
         investigatorId.setIdSource("investigatorId");
         investigatorId.setSampleId(sampleManifestEntity.getInvestigatorSampleId());
-        sampleManifestEntity.addSample(investigatorId);
+        metaDbSample.addSample(investigatorId);
 
-        SampleManifestJsonEntity sampleJson = new SampleManifestJsonEntity();
-        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
-        sampleJson.setCreationDate(String.valueOf(timestamp.getTime()));
-        sampleJson.setSampleManifestJson(sampleManifestEntity.toString());
-        sampleManifestEntity.setSampleManifestJsonEntity(sampleJson);
-
-        return sampleManifestEntity;
+        return metaDbSample;
+    }
+    
+    @Override
+    public MetaDbSample setUpSampleManifestEntity(MetaDbSample metaDbSample) throws Exception {
+        for (SampleManifestEntity s: metaDbSample.getSampleManifestList()) {
+            s.setSampleManifestJson(s.toString());
```

Comment:
are we sure that `s.toString()` is going to be formatting the json string correctly? We should convert to a json string with gson instead